test/tests/transact.ts

```diff
@@ -192,55 +135,133 @@ suite('transact', function () {
+                const result = await session.transact(
+                    {action},
+                    {
+                        allowModify: true,
+                        transactPlugins: [new MockTransactResourceProviderPlugin()],
+                    }
+                )
                 assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 2)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
             })
             test('false', async function () {
-                const result = await session.transact({action}, {allowModify: false})
-                assetValidTransactResponse(result)
-            })
-            test('ignores modification from beforeSign hooks', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact(
                     {action},
                     {
                         allowModify: false,
-                        hooks: {
-                            beforeSign: [new MockTransactHook()],
-                        },
+                        transactPlugins: [new MockTransactResourceProviderPlugin()],
                     }
                 )
                 assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 1)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
             })
         })
         suite('broadcast', function () {
-            test('undefined', async function () {
+            test('default: true', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact({action})
                 assetValidTransactResponse(result)
             })
             test('true', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact({action}, {broadcast: true})
                 assetValidTransactResponse(result)
             })
             test('false', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact({action}, {broadcast: false})
                 assetValidTransactResponse(result)
             })
         })
+        suite('transactPlugins', function () {
+            test('inherit', async function () {
+                const {action} = await mockData()
+                const session = new Session({
+                    ...mockSessionOptions,
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
```

Comment:
And if manually assembling a `Session` object, the `TransactPlugin` can be passed in during construction.